### ISSUE TITLE
Floor js json import enters infinite loop of import warnings and errors

### ISSUE THREAD

**Issue by antonszilasi** [NREL/OpenStudio#4237]
Issue overview
--------------
Importing the floor js json here https://drive.google.com/file/d/1CPfOm_62G1KLVPXrRjkPa1B6FiaaU-JB/view?usp=sharing

Causes OpenStudio to enter an infinite loop of import warnings and errors

Im pretty sure that this is because the geometry isn't the cleanest maybe this kind of geometry shouldnt be supported?

![image](https://user-images.githubusercontent.com/8773483/110411636-27489180-8048-11eb-9fc7-8f4aa8c49a5a.png)

### Environment

OpenStudio 3.1.0

**Comment by antonszilasi** [comment]
Note de-drawing this geometry with the correct grid alignment so that it is straight works fine

![image](https://user-images.githubusercontent.com/8773483/110541110-8cea5b80-80e4-11eb-997e-c7ff01c27670.png)

**Comment by jmarrec** [comment]
Extract:

```
[openstudio.model.Surface] <-1> Intersection of 'Surface 485' with 'Surface 9' results in (Surface 485), (Surface 9), [Surface 486], []
^C[openstudio.model.Surface] <1> Initial area of surface 'Surface 486' 0.94743 does not equal post intersection area 0.958578
[openstudio.model.Surface] <-1> Intersection of 'Surface 486' with 'Surface 10' results in (Surface 486), (Surface 10), [Surface 487], []
[utilities.geometry.intersect] <-1> Cannot compute area of face difference, result will not include this polygon, []
[openstudio.model.Surface] <-1> Intersection of 'Surface 487' with 'Surface 9' results in (Surface 487), (Surface 9), [Surface 488], []
[openstudio.model.Surface] <1> Initial area of surface 'Surface 488' 0.94743 does not equal post intersection area 0.958578
[openstudio.model.Surface] <-1> Intersection of 'Surface 488' with 'Surface 10' results in (Surface 488), (Surface 10), [Surface 489], []
^C[utilities.geometry.intersect] <-1> Cannot compute area of face difference, result will not include this polygon, []
[openstudio.model.Surface] <-1> Intersection of 'Surface 489' with 'Surface 9' results in (Surface 489), (Surface 9), [Surface 490], []
^C[openstudio.model.Surface] <1> Initial area of surface 'Surface 490' 0.94743 does not equal post intersection area 0.958578
^C[openstudio.model.Surface] <-1> Intersection of 'Surface 490' with 'Surface 10' results in (Surface 490), (Surface 10), [Surface 491], []
^C[utilities.geometry.intersect] <-1> Cannot compute area of face difference, result will not include this polygon, []
[openstudio.model.Surface] <-1> Intersection of 'Surface 491' with 'Surface 9' results in (Surface 491), (Surface 9), [Surface 492], []
^C[openstudio.model.Surface] <1> Initial area of surface 'Surface 492' 0.94743 does not equal post intersection area 0.958578
[openstudio.model.Surface] <-1> Intersection of 'Surface 492' with 'Surface 10' results in (Surface 492), (Surface 10), [Surface 493], []
^C[utilities.geometry.intersect] <-1> Cannot compute area of face difference, result will not include this polygon, []
^C[openstudio.model.Surface] <-1> Intersection of 'Surface 493' with 'Surface 9' results in (Surface 493), (Surface 9), [Surface 494], []
^C[openstudio.model.Surface] <1> Initial area of surface 'Surface 494' 0.94743 does not equal post intersection area 0.958578
[openstudio.model.Surface] <-1> Intersection of 'Surface 494' with 'Surface 10' results in (Surface 494), (Surface 10), [Surface 495], []

```

**Comment by tijcolem** [comment]
@jmarrec Thanks for testing. Now that 3.2.0 is out with all the geo updates, can we re-test this on 3.2.0 and see if the issue is still present?

**Comment by jmarrec** [comment]
Fixed in 3.2.0, via #4221. I can properly rt the json supplied.

```ruby
json = 'issue_4237.json'
floorplan = OpenStudio::FloorplanJS.load(json)
scene = floorplan.get.toThreeScene(true)
rt = OpenStudio::Model::ThreeJSReverseTranslator.new
model = rt.modelFromThreeJS(scene)
```

```ruby
[openstudio.model.Surface] <1> Initial area of surface 'Face 157' 5.48895 does not equal post intersection area 5.4799
[openstudio.model.Surface] <1> Initial area of surface 'Surface 3' 6.91108 does not equal post intersection area 6.90437
[openstudio.model.Surface] <1> Initial area of other surface 'Face 359' 56.5808 does not equal post intersection area 56.5826
[openstudio.model.Surface] <1> Initial area of surface 'Surface 5' 0.00880208 does not equal post intersection area 0.00511317
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 8' 53.0151 does not equal post intersection area 53.008
[openstudio.model.Surface] <1> Initial area of surface 'Face 39' 11.9756 does not equal post intersection area 11.9583
[openstudio.model.Surface] <1> Initial area of surface 'Surface 10' 4.91798 does not equal post intersection area 4.90116
[openstudio.model.Surface] <1> Initial area of surface 'Face 134' 15.6851 does not equal post intersection area 15.6684
[openstudio.model.Surface] <1> Initial area of other surface 'Face 395' 10.5631 does not equal post intersection area 10.5463
[openstudio.model.Surface] <1> Initial area of surface 'Surface 14' 15.5519 does not equal post intersection area 15.546
[openstudio.model.Surface] <1> Initial area of other surface 'Face 424' 35.8894 does not equal post intersection area 35.8493
[openstudio.model.Surface] <1> Initial area of surface 'Surface 16' 0.0467116 does not equal post intersection area 0.0415539
[openstudio.model.Surface] <1> Initial area of other surface 'Face 348' 37.2597 does not equal post intersection area 37.1554
[openstudio.model.Surface] <1> Initial area of surface 'Surface 20' 10.6463 does not equal post intersection area 10.6439
[openstudio.model.Surface] <1> Initial area of surface 'Surface 22' 10.1347 does not equal post intersection area 10.1188
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 9' 53.0029 does not equal post intersection area 52.9987
[openstudio.model.Surface] <1> Initial area of other surface 'Face 334' 35.6916 does not equal post intersection area 35.6868
[openstudio.model.Surface] <1> Initial area of surface 'Face 121' 29.5432 does not equal post intersection area 29.546
[openstudio.model.Surface] <1> Initial area of surface 'Face 165' 43.0438 does not equal post intersection area 43.0314
[openstudio.model.Surface] <1> Initial area of other surface 'Face 481' 23.8656 does not equal post intersection area 23.8532
[openstudio.model.Surface] <1> Initial area of other surface 'Face 448' 75.4373 does not equal post intersection area 75.3872
[openstudio.model.Surface] <1> Initial area of surface 'Surface 58' 48.9326 does not equal post intersection area 48.9039
[openstudio.model.Surface] <1> Initial area of other surface 'Face 280' 58.0838 does not equal post intersection area 58.0552
[openstudio.model.Surface] <1> Initial area of surface 'Face 1' 59.2721 does not equal post intersection area 59.1653
[openstudio.model.Surface] <1> Initial area of surface 'Face 179' 70.4602 does not equal post intersection area 70.4472
[openstudio.model.Surface] <1> Initial area of other surface 'Face 471' 13.2956 does not equal post intersection area 13.2826
[openstudio.model.Surface] <1> Initial area of surface 'Face 102' 109.626 does not equal post intersection area 119.096
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 18' 19.8408 does not equal post intersection area 19.7626
[utilities.geometry.intersect] <1> Face difference is self intersecting, result will not include this polygon, [[6.59007, 3.05816, 0], [7.28095, 4.2672, 0], [9.10975, 7.9248, 0], [7.2198, 4.14489, 0]]
[openstudio.model.Surface] <1> Initial area of surface 'Surface 78' 78.7841 does not equal post intersection area 78.8167
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 30' 37.1105 does not equal post intersection area 37.1052
[openstudio.model.Surface] <1> Initial area of surface 'Surface 81' 25.6052 does not equal post intersection area 25.5517
[openstudio.model.Surface] <1> Initial area of other surface 'Face 313' 52.2115 does not equal post intersection area 52.2168
[openstudio.model.Surface] <1> Initial area of surface 'Face 396' 10.5631 does not equal post intersection area 10.5495
[openstudio.model.Surface] <1> Initial area of other surface 'Face 667' 26.8425 does not equal post intersection area 26.8289
[openstudio.model.Surface] <1> Initial area of surface 'Surface 86' 10.3512 does not equal post intersection area 10.3424
[openstudio.model.Surface] <1> Initial area of other surface 'Face 677' 48.4407 does not equal post intersection area 48.4307
[openstudio.model.Surface] <1> Initial area of surface 'Surface 88' 0.0982394 does not equal post intersection area 0.0922537
[openstudio.model.Surface] <1> Initial area of other surface 'Face 716' 110.013 does not equal post intersection area 110.003
[openstudio.model.Surface] <1> Initial area of surface 'Face 414' 11.3342 does not equal post intersection area 11.315
[openstudio.model.Surface] <1> Initial area of other surface 'Face 709' 7.53095 does not equal post intersection area 7.51992
[openstudio.model.Surface] <1> Initial area of other surface 'Face 695' 72.0411 does not equal post intersection area 72.0421
[openstudio.model.Surface] <1> Initial area of surface 'Surface 99' 0.0941594 does not equal post intersection area 0.0831204
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 102' 68.0664 does not equal post intersection area 68.049
[openstudio.model.Surface] <1> Initial area of surface 'Face 472' 13.2956 does not equal post intersection area 13.2707
[openstudio.model.Surface] <1> Initial area of other surface 'Face 805' 42.9154 does not equal post intersection area 42.9241
[openstudio.model.Surface] <1> Initial area of surface 'Face 541' 13.6155 does not equal post intersection area 13.552
[openstudio.model.Surface] <1> Initial area of other surface 'Face 851' 38.2439 does not equal post intersection area 38.1937
[openstudio.model.Surface] <1> Initial area of surface 'Face 249' 16.1963 does not equal post intersection area 16.1498
[openstudio.model.Surface] <1> Initial area of other surface 'Face 570' 86.4074 does not equal post intersection area 86.3905
[openstudio.model.Surface] <1> Initial area of surface 'Surface 109' 13.9452 does not equal post intersection area 13.9258
[openstudio.model.Surface] <1> Initial area of surface 'Face 482' 23.8656 does not equal post intersection area 23.821
[openstudio.model.Surface] <1> Initial area of surface 'Face 527' 24.2889 does not equal post intersection area 24.2655
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 106' 24.7619 does not equal post intersection area 24.7277
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 108' 70.5658 does not equal post intersection area 70.5271
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 103' 67.9659 does not equal post intersection area 67.9595
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 91' 14.873 does not equal post intersection area 14.8645
[openstudio.model.Surface] <1> Initial area of surface 'Face 335' 35.6916 does not equal post intersection area 35.6865
[openstudio.model.Surface] <1> Initial area of surface 'Surface 136' 28.722 does not equal post intersection area 28.7063
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 116' 24.7694 does not equal post intersection area 24.754
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 112' 47.8693 does not equal post intersection area 47.8329
[openstudio.model.Surface] <1> Initial area of surface 'Face 425' 35.8894 does not equal post intersection area 35.8796
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 87' 26.6306 does not equal post intersection area 26.61
[openstudio.model.Surface] <1> Initial area of surface 'Surface 141' 9.55852 does not equal post intersection area 9.5544
[openstudio.model.Surface] <1> Initial area of other surface 'Face 636' 35.8544 does not equal post intersection area 35.828
[openstudio.model.Surface] <1> Initial area of surface 'Surface 142' 0.00523408 does not equal post intersection area 0.00185914
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 89' 0.00516583 does not equal post intersection area 0.00185914
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 126' 43.3388 does not equal post intersection area 43.3061
[openstudio.model.Surface] <1> Initial area of surface 'Face 349' 37.2597 does not equal post intersection area 37.1407
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 143' 0.294143 does not equal post intersection area 0.274995
[openstudio.model.Surface] <1> Initial area of surface 'Surface 146' 36.8657 does not equal post intersection area 36.8473
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 139' 16.6634 does not equal post intersection area 16.6507
[openstudio.model.Surface] <1> Initial area of surface 'Surface 147' 27.8249 does not equal post intersection area 27.8004
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 144' 26.2736 does not equal post intersection area 26.0193
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 90' 38.1813 does not equal post intersection area 38.1726
[openstudio.model.Surface] <1> Initial area of surface 'Face 507' 42.3638 does not equal post intersection area 42.3128
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 125' 0.677956 does not equal post intersection area 0.644113
[openstudio.model.Surface] <1> Initial area of surface 'Surface 154' 41.6687 does not equal post intersection area 41.6666
[openstudio.model.Surface] <1> Initial area of other surface 'Face 836' 42.6057 does not equal post intersection area 42.5456
[openstudio.model.Surface] <1> Initial area of surface 'Face 314' 52.2115 does not equal post intersection area 52.1645
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 151' 25.2269 does not equal post intersection area 25.1558
[openstudio.model.Surface] <1> Initial area of surface 'Surface 157' 27.2809 does not equal post intersection area 27.237
[openstudio.model.Surface] <1> Initial area of surface 'Face 360' 56.5808 does not equal post intersection area 56.5721
[openstudio.model.Surface] <1> Initial area of surface 'Surface 160' 49.3042 does not equal post intersection area 49.293
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 129' 61.9023 does not equal post intersection area 61.8877
[openstudio.model.Surface] <1> Initial area of surface 'Surface 161' 0.11381 does not equal post intersection area 0.104675
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 131' 28.1532 does not equal post intersection area 28.1388
[openstudio.model.Surface] <1> Initial area of surface 'Face 281' 58.0838 does not equal post intersection area 58.03
[openstudio.model.Surface] <1> Initial area of surface 'Surface 170' 42.799 does not equal post intersection area 42.7569
[openstudio.model.Surface] <1> Initial area of surface 'Surface 171' 22.7205 does not equal post intersection area 22.4238
[openstudio.model.Surface] <1> Initial area of surface 'Surface 172' 19.3919 does not equal post intersection area 19.3809
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 169' 28.0341 does not equal post intersection area 28.0329
[openstudio.model.Surface] <1> Initial area of surface 'Surface 178' 6.76337 does not equal post intersection area 6.73507
[openstudio.model.Surface] <1> Initial area of surface 'Surface 186' 67.9569 does not equal post intersection area 67.2042
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 145' 6.66387 does not equal post intersection area 5.85363
[openstudio.model.Surface] <1> Initial area of other surface 'Face 554' 172.8 does not equal post intersection area 172.772
[openstudio.model.Surface] <1> Initial area of other surface 'Face 797' 17.7654 does not equal post intersection area 17.7411
[openstudio.model.Surface] <1> Initial area of surface 'Surface 194' 45.4409 does not equal post intersection area 45.4425
[openstudio.model.Surface] <1> Initial area of other surface 'Face 770' 46.7854 does not equal post intersection area 46.7084
[openstudio.model.Surface] <1> Initial area of other surface 'Face 815' 42.0505 does not equal post intersection area 42.0288
[openstudio.model.Surface] <1> Initial area of surface 'Surface 200' 41.5668 does not equal post intersection area 41.516
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 156' 0.896926 does not equal post intersection area 0.834599
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 123' 5.95642 does not equal post intersection area 5.94764
[openstudio.model.Surface] <1> Initial area of other surface 'Face 826' 43.4413 does not equal post intersection area 43.4181
[openstudio.model.Surface] <1> Initial area of surface 'Face 203' 196.583 does not equal post intersection area 196.571
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 101' 0.274824 does not equal post intersection area 0.264318
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 185' 28.4361 does not equal post intersection area 28.423
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 187' 45.0496 does not equal post intersection area 45.0389
[openstudio.model.Surface] <1> Initial area of other surface 'Surface 192' 127.452 does not equal post intersection area 127.309
=> #<OpenStudio::Model::OptionalModel:0x000055d01df7d1e8 @__swigtype__="_p_boost__optionalT_openstudio__model__Model_t">
```

```
[8] 4237_jsonInfiniteLoop(main)> model.get
=> #<OpenStudio::Model::Model:0x000055d01e4d8110 @__swigtype__="_p_openstudio__model__Model">
```ruby

**Comment by antonszilasi** [comment]
@jmarrec great work will try on my end
